glean-core/python/glean/_process_dispatcher.py

```diff
@@ -0,0 +1,76 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+This module implements a class to dispatch work to another process.
+This work always happens one-at-a-time.  That is, it doesn't create
+another worker process until the previous worker process has completed.
+
+This is used only by the PingUploadWorker (and the test-only feature to delete
+temporary data directories which has a potential race condition with the
+PingUploadWorker).
+"""
+
+import sys
+from typing import Optional, TYPE_CHECKING, Union
+
+
+if TYPE_CHECKING:
+    import multiprocessing  # noqa
```

Comment:
Probably some python thingies, but why is this behind an if?

Reply:
This is the nexus of a bunch of half solutions in Python ;)  Importing the multiprocessing module could fail in contexts where it won't work (the one we care about is PyInstaller which mozregression uses to create a single-file distribution).  So we put it behind a config flag (`allow_multiprocessing`), but we still want to do type-checking (at build time) on things that involve multiprocessing, so we have to unconditionally import it so the type checking tool (mypy) can do it's thing.